Marlin/src/module/motion.cpp

```diff
@@ -1473,7 +1473,7 @@ void homeaxis(const AxisEnum axis) {
   #endif
 } // homeaxis()
 
-#if HAS_WORKSPACE_OFFSET || ENABLED(DUAL_X_CARRIAGE) || ENABLED(DELTA)
+#if HAS_WORKSPACE_OFFSET || ENABLED(DUAL_X_CARRIAGE) || ENABLED(DELTA) || HAS_SOFTWARE_ENDSTOPS
```

Comment:
I still disagree with this change. Simply having software endstops is not enough to justify updating them. There must be some change possible, which is what the other conditions imply. That said, these other conditions are probably a little out of date. When I converted Marlin to use native machine coordinates, this meant that the soft endstops —which are also expressed in native machine coordinates— no longer needed to be updated merely due to a workspace offset. I think the only conditions now are DELTA and Dual X Carriage. I will double-check and add a patch soon.

Reply:
> I think the only conditions now are DELTA and Dual X Carriage. I will double-check and add a patch soon.

Just as an FYI.   IDEX is kind of special with regard to software endstops.   The right carriage needs to move all the way off the bed on the right side.  And the left carriage needs to move all the way off the bed on the left side.   So...   in a sense, there are two different software endstops that should be enforced for the X-Axis.

Reply:
And some of those idex soft and stop conditions were the point of a few of my earlier PRS, as initially there were plenty of ways that the heads could crash into each other.